pkg/tsdb/azuremonitor/types/types.go

```diff
@@ -213,3 +213,7 @@ type LogAnalyticsWorkspaceResponse struct {
 	PublicNetworkAccessForQuery     string                          `json:"publicNetworkAccessForQuery"`
 	RetentionInDays                 int                             `json:"retentionInDays"`
 }
+
+type AzureError interface {
+	Error() string
+}
```

Comment:
You can define this as an error (I would also name the error something more specific):

```
var AzureHealthCheckError = errors.New("health check failed")
```

You can see this approach many times in our [code base](https://github.com/grafana/grafana/blob/main/pkg/models/org.go#L10)